.github/workflows/bump-deps.yaml

```diff
@@ -0,0 +1,88 @@
+# Description: Run this by visiting https://github.com/brave/brave-core/actions/workflows/bump-deps.yaml
+# and clicking the `Run workflow` button on the top-right.
+# You can also run it from cli by: `gh workflow run bump-deps.yaml [-f dep=vendor/web-discovery-project] [-f ref=commit-hash]`
+# If `ref` is not given, we will try to use the latest commmit (tip) of a branch called `master` or `main`
+name: Bump DEPS
+on:
+  workflow_dispatch:
+    inputs:
+      dep:
+        description: "Dependency"
+        required: true
+        type: string
+        default: "vendor/web-discovery-project"
+      ref:
+        description: "New ref for dependency"
+        required: false
+        type: string
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  auto-bump:
+    runs-on: ubuntu-latest
+    env:
+      GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: "0"
+
+      - name: auto-bump
+        run: |
+          shopt -s inherit_errexit
+          set -eEo pipefail
+
+          SED=sed
+
+          # Inputs
+          DEP="${{ inputs.dep }}"
+          DEPREF="${{ inputs.ref }}"
+
+          [[ "${DEP:?}" =~ ^[a-zA-Z/-]+$ ]] || { echo Input dependency validation failed; exit 1; }
+          if [[ -z "${DEPREF}" ]]; then
+            echo "Attempting to guess ref hash"
+            DEP_REPO=$(\
+                sed -n '/deps = {/, /^}/p' DEPS|\
+                sed 's/deps = //'|\
+                python3 -c "import sys,ast; d=ast.literal_eval(sys.stdin.read()); dep=d[\""$DEP"\"]; print(dep) if isinstance(dep, str) else print(dep.get(\"url\"))"|\
+                grep -Po '(?<=github.com\/).*?(?=\.git)?(?=@)'|\
+                sed 's/\.git//')
+            gh api /repos/"${DEP_REPO}"/commits/master -q .sha > /tmp/DEPREF || gh api /repos/"${DEP_REPO}"/commits/main -q .sha > /tmp/DEPREF
+            DEPREF=$(</tmp/DEPREF)
+          fi
+          [[ "${DEPREF:?}" =~ ^[a-z0-9]+$ ]] || { echo Dependency reference validation failed; exit 1; }
+          ! $(grep -q "$DEPREF" DEPS) || { echo Dependency reference alread exists in DEPS. Cannot continue; exit 1; }
+
+          # readarray -t milestones < <(gh api /repos/${GITHUB_REPOSITORY:?}/milestones -q ".[] | .title" | grep -Fv Closed)
```

Comment:
To remove comments.

Reply:
I'd rather keep them to be able to uncomment easily later on if needed